### PR TITLE
Make expression mining use configurable options and logic

### PR DESCRIPTION
**Pull request by ajreynol**
Required for doing options-specific internal fuzzing using SyGuS.

### CODE REVIEW

File: src/theory/quantifiers/expr_miner.cpp [hunk 1]
```diff
@@ -52,18 +52,24 @@ Node ExprMiner::convertToSkolem(Node n)
...
 {
   Assert (!query.isNull());
   if (Options::current().quantifiers.sygusExprMinerCheckTimeoutWasSetByUser)
   {
-    initializeSubsolver(checker,
-                        d_env,
-                        true,
-                        options::sygusExprMinerCheckTimeout());
+    initializeSubsolver(
+        checker, opts, logicInfo, true, options::sygusExprMinerCheckTimeout());
```
Comment (4tXJ7f) [inline src/theory/quantifiers/expr_miner.cpp:68]: It feels a bit dangerous that if the expression miner check timeout is set, the passed options are ignored.

**Review by 4tXJ7f — APPROVED** [review]
LGTM, I just have a concern that we can address in a future PR if needed.

File: src/theory/quantifiers/expr_miner.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/quantifiers/expr_miner.cpp:68]: What do you mean? We do carry `opts`, `logicInfo` here?

File: src/theory/quantifiers/expr_miner.cpp [hunk 2]
```diff
@@ -52,18 +52,24 @@ Node ExprMiner::convertToSkolem(Node n)
...
+  initializeChecker(checker, query, options(), logicInfo());
+}
+
+void ExprMiner::initializeChecker(std::unique_ptr<SolverEngine>& checker,
+                                  Node query,
+                                  const Options& opts,
+                                  const LogicInfo& logicInfo)
 {
   Assert (!query.isNull());
   if (Options::current().quantifiers.sygusExprMinerCheckTimeoutWasSetByUser)
```
Comment (nafur) [inline src/theory/quantifiers/expr_miner.cpp:65]: Can we use the `opts` object here instead of using `options::` and `Options::current()`.

Comment (ajreynol) [reply src/theory/quantifiers/expr_miner.cpp:65]: Yeah, I'll clean this on a separate PR, this is old.

**Pull request merged**